neqo-transport/src/recv_stream.rs

```diff
@@ -200,41 +200,46 @@ impl RxStreamOrderer {
             false
         };
 
-        // Now handle possible overlap with next entries
-        let mut to_remove = SmallVec::<[_; 8]>::new();
         let mut to_add = new_data;
+        if let Some(entry) = self.data_ranges.last_entry() {
+            // is this at the end (common case)?  If so, don't need this block
+            if *entry.key() > new_start {
+                // Now handle possible overlap with next entries
+                let mut to_remove = SmallVec::<[_; 8]>::new();
+
+                for (&next_start, next_data) in self.data_ranges.range_mut(new_start..) {
+                    let next_end = next_start + u64::try_from(next_data.len()).unwrap();
+                    let overlap = new_end.saturating_sub(next_start);
+                    if overlap == 0 {
+                        break;
+                    } else if next_end >= new_end {
```

Comment:
No else after break...  However, the overlap == 0 block could be removed.  The next block is a noop when overlap == 0.